agent/stats/engine_unix_integ_test.go

```diff
@@ -32,3 +55,96 @@ func TestStatsEngineWithNetworkStatsDifferentModes(t *testing.T) {
 		testNetworkModeStats(t, tc.NetworkMode, tc.StatsEmpty)
 	}
 }
+
+func TestStatsEngineWithServiceConnectMetrics(t *testing.T) {
+	// Create a new docker stats engine
+	engine := NewDockerStatsEngine(&cfg, dockerClient, eventStream("TestStatsEngineWithExistingContainersWithoutHealth"))
+	ctx, cancel := context.WithCancel(context.TODO())
+	defer cancel()
+
+	// Assign ContainerStop timeout to addressable variable
+	timeout := defaultDockerTimeoutSeconds
+
+	// Create a container to get the container id.
+	container, err := createGremlin(client, "default")
+	require.NoError(t, err, "creating container failed")
+	defer client.ContainerRemove(ctx, container.ID, types.ContainerRemoveOptions{Force: true})
+
+	engine.cluster = defaultCluster
+	engine.containerInstanceArn = defaultContainerInstance
+
+	err = client.ContainerStart(ctx, container.ID, types.ContainerStartOptions{})
+	require.NoError(t, err, "starting container failed")
+	defer client.ContainerStop(ctx, container.ID, &timeout)
+
+	containerChangeEventStream := eventStream("TestStatsEngineWithExistingContainersWithoutHealth")
+	taskEngine := ecsengine.NewTaskEngine(&config.Config{}, nil, nil, containerChangeEventStream,
+		nil, dockerstate.NewTaskEngineState(), nil, nil, nil)
+	testTask := createRunningTask()
+	testTask.ServiceConnectConfig = &apitask.ServiceConnectConfig{
+		ContainerName: serviceConnectContainerName,
+		RuntimeConfig: apitask.RuntimeConfig{
+			AdminSocketPath: testUDSPath,
+			StatsRequest:    testStatsRestPath,
```

Comment:
This needs to be updated since an open PR #3224 has changes to remove `StatsRequest` from  RuntimeConfig